### ISSUE TITLE
[Linux] Chat wheel always goes left when playing on 2nd monitor

### ISSUE THREAD

**Issue by zjeffer** [ValveSoftware/Dota-2#1448]
#### Your system information

* System information: https://gist.github.com/zjeffer/6ffc4863e3218516355316e4a09b4dc5
* Have you checked for system updates?: [Yes]
* Are you using the latest stable video driver available for your system? [Yes]
* Have you verified the game files?: [Yes]

#### Please describe your issue in as much detail as possible:
When starting the game with default window settings (which is using desktop-friendly fullscreen mode) the game starts on my laptop monitor. My second monitor is bigger, so I move the window using alt-click to my second monitor once the game boots up. When in-game, I can't use the chatwheel because the cursor immediately goes to the left, choosing the left-most option. Strangely enough, it also moves the whole screen to the left because it thinks the cursor is at the edge of the screen. 
This happens only in desktop-friendly fullscreen mode when playing on the second monitor. 
When I set it to windowed mode, everything works fine. When I set it back to desktop-friendly fullscreen mode, it somehow works fine as well, until I restart the game. 

Using these launch options:
vblank_mode=0 primusrun %command%
This is because I'm using the bumblebee driver for my Nvidia GPU (latest version)
Using Arch Linux kernel 4.16.3-1 with KDE

#### Steps for reproducing this issue:

1. Start the game with desktop-friendly mode enabled (for me it's the default option)
2. Move the window to the second screen (for me it's configured on my right side)
3. Join a game/demo mode
4. Use the chatwheel and your view will go left and it will choose the left-most option.

#### To fix:
Either never use desktop-friendly fullscreen mode.
OR
Use desktop friendly fullscreen mode, but everytime you play, go to settings and change it to windowed, and then back to desktop-friendly fullscreen mode. That works for me until the next restart.

So basically when I exit the game from windowed mode and start it again and then choose desktop-friendly fullscreen, it will work fine. The problem happens when launching the game with desktop-friendly fullscreen mode enabled.

This is the same issue as #198 
EDIT: and #1201 , but maybe mine has some more info to help (finally) fix this.

I hope that's enough info.

**Comment by gdrewb-valve** [comment]
If you start on the right window (use -sdl_displayindex 1) does that fix it?

**Comment by zjeffer** [comment]
Thanks for the quick answer. 
How do I add that to my launch options?
Right now I changed it to this: -sdl_displayindex 1 vblank_mode=0 primusrun %command% 
But this doesn't move the window.

**Comment by kisak-valve** [comment]
When using `%command%` in the launch options, game arguments must go after %command%.

In your case that would be `vblank_mode=0 primusrun %command% -sdl_displayindex 1`.

**Comment by zjeffer** [comment]
Thanks. 
So now, the game still opens on my laptop monitor, and before moving it, the chatwheel goes to the right instead (which is expected i guess, because it's pointing at the chosen monitor). So I move it to my right monitor, which is the one I want to use, and that works!

I did some more testing and found that when I unplug my second monitor, the chatwheel works fine and doesn't point to the right, because there isn't a monitor there anymore. This is useful because I don't always have access to a secondary monitor so I can also play without issues on my laptop screen. 

I guess the only issue that's left is if some people like to switch between monitors from time to time for whatever reason, they'll still have this issue. That doesn't bother me however so I guess you can close this issue if you want to.

Thanks for all the help!

**Comment by Snooober** [comment]
The latest update (invoker persona...🤮) seems to have fixed this issue for me on Solus OS.

EDIT: Spoke too soon still having the bug. Not sure why it was working briefly.

EDIT2: It works sometimes. I can't figure out when. It attaches the mouse to my left or right monitor even though DotA launches on my left monitor every time.

**Comment by mhavasi** [comment]
I also have this bug. Playing on linux PC with only one screen. I have no idea how to solve it.

**Comment by zjeffer** [comment]
With only one screen? What's the output of `xrandr`?

**Comment by mhavasi** [comment]
Screen 0: minimum 8 x 8, current 2560 x 1440, maximum 32767 x 32767
DP-0 disconnected (normal left inverted right x axis y axis)
DP-1 disconnected (normal left inverted right x axis y axis)
DP-2 connected primary 2560x1440+0+0 (normal left inverted right x axis y axis) 610mm x 356mm
   2560x1440     59.95*+  59.95  
   1920x1200     59.88  
   1920x1080     60.00    59.94    50.00    23.98  
   1680x1050     59.95  
   1600x900      60.00  
   1440x900      59.89  
   1280x1024     60.02  
   1280x800      59.81  
   1280x720      60.00    59.94    50.00  
   1024x768      60.00  
   800x600       60.32  
   720x576       50.00  
   720x480       59.94  
   640x480       59.94    59.93  
DP-3 disconnected (normal left inverted right x axis y axis)
DP-4 disconnected (normal left inverted right x axis y axis)
DP-5 disconnected (normal left inverted right x axis y axis)
DP-6 disconnected (normal left inverted right x axis y axis)
DP-7 disconnected (normal left inverted right x axis y axis)

**Comment by AHolliday** [comment]
Just chiming in to say I have the same issue as mhavasi.  I'm playing on Linux Mint 19.1, with just one display connected.

**Comment by Lambdara** [comment]
Same issue here, on Ubuntu 20.04.1; using Gnome. Setting the screen to windowed mode, moving it to the correct monitor, and then going to fullscreen also fixes it for me until next restart.
